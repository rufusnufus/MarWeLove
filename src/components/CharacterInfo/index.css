.characterinfo {
  display: flex;
  color: white;
  margin-bottom: 130px;
  width: 80%;
  margin: 100px auto;
  justify-content: space-evenly;
  max-height: 500px;
}

.characterinfo__name {
  font-weight: bold;
  font-size: 50px;
  line-height: 74px;
}

.characterinfo__text {
  width: 50%;
  max-height: 100%;
  overflow: auto;
}

.characterinfo__description {
  margin-top: 20px;
  font-weight: bold;
  font-size: 25px;
  width: 100%;
}

.characterinfo__image {
  width: 463px;
  height: 463px;
  background-size: cover;
}
