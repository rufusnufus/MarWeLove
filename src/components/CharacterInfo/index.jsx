import React from 'react'

import './index.css'

function CharacterInfo({ data }) {
  return (
    <div className="characterinfo">
      <div className="characterinfo__text">
        <div className="characterinfo__name">{data.caption}</div>
        <div className="characterinfo__description">{data.description}</div>
      </div>
      <div className="characterinfo__image" style={{ backgroundImage: `url(${data.imageUrl})` }} />
    </div>
  )
}

export default CharacterInfo
