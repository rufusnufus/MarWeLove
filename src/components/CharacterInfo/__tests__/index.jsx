/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render } from '@testing-library/react'
import CharacterInfo from '../index'

afterEach(cleanup)

const mockData = {
  id: 1009144,
  name: 'Abyss (Age of Apocalypse)',
  description: 'AIM is a terrorist organization bent on destroying the world.',
  thumbnail: {
    path: 'http://i.annihil.us/u/prod/marvel/i/mg/6/20/52602f21f29ec',
    extension: 'jpg'
  }
}

describe('CharacterInfo', () => {
  it('Character Name', () => {
    const { queryByText } = render(<CharacterInfo data={mockData} />)

    expect(queryByText('Abyss (Age of Apocalypse)')).toBeTruthy()
  })

  it('Character Description', () => {
    const { queryByText } = render(<CharacterInfo data={mockData} />)

    expect(queryByText('AIM is a terrorist organization bent on destroying the world.')).toBeTruthy()
  })
})
