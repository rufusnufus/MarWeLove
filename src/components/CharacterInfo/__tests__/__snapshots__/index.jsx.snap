// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfo Render with thumbnail 1`] = `
<div>
  <div
    class="characterinfo"
  >
    <div
      class="characterinfo__text"
    >
      <div
        class="bookmark__container"
      >
        <div
          class="characterinfo__name"
        >
          Abyss (Age of Apocalypse)
        </div>
        Bookmark
      </div>
      <div
        class="characterinfo__description"
      >
        AIM is a terrorist organization bent on destroying the world.
      </div>
    </div>
    <div
      class="characterinfo__image"
      style="background-image: url(.);"
    />
  </div>
</div>
`;

exports[`CharacterInfo Render without thumbnail 1`] = `
<div>
  <div
    class="characterinfo"
  >
    <div
      class="characterinfo__text"
    >
      <div
        class="bookmark__container"
      >
        <div
          class="characterinfo__name"
        >
          Abyss (Age of Apocalypse)
        </div>
        Bookmark
      </div>
      <div
        class="characterinfo__description"
      >
        AIM is a terrorist organization bent on destroying the world.
      </div>
    </div>
    <div
      class="characterinfo__image"
      style="background-image: url(undefined.undefined);"
    />
  </div>
</div>
`;
